tests/test_settings.py

```diff
@@ -1110,6 +1141,55 @@ def test_connection_string_with_dollar_sign(self):
             assert url.database == "the-database"
             assert url.password == "the-$password"
 
+    def test_sqlalchemy_settings_migration(self):
+        """Test that SQLAlchemy settings work with both old and new structures."""
+
+        with pytest.warns(
+            DeprecationWarning, match="moved to the `sqlalchemy` settings group."
+        ):
+            settings_with_old_keys = Settings(
+                server=ServerSettings(
+                    database=ServerDatabaseSettings(
+                        sqlalchemy_pool_size=42,
+                        sqlalchemy_max_overflow=37,
+                    )
+                )
+            )
+            assert settings_with_old_keys.server.database.sqlalchemy_pool_size == 42
+            assert settings_with_old_keys.server.database.sqlalchemy_max_overflow == 37
+
+        assert settings_with_old_keys.server.database.sqlalchemy.pool_size == 42
+        assert settings_with_old_keys.server.database.sqlalchemy.max_overflow == 37
+
+        settings_with_new_keys = Settings(
+            server=ServerSettings(
+                database=ServerDatabaseSettings(
+                    sqlalchemy=SQLAlchemySettings(
+                        pool_size=42,
+                        max_overflow=37,
+                    )
+                )
+            )
+        )
+
+        with pytest.warns(
+            DeprecationWarning, match="moved to the `sqlalchemy` settings group."
+        ):
+            assert settings_with_new_keys.server.database.sqlalchemy_pool_size == 42
+            assert settings_with_new_keys.server.database.sqlalchemy_max_overflow == 37
+
+        # new keys are updated by setting new keys
+        assert settings_with_new_keys.server.database.sqlalchemy.pool_size == 42
+        assert settings_with_new_keys.server.database.sqlalchemy.max_overflow == 37
+
+    def test_sqlalchemy_settings_migration_via_env_var(
```

Comment:
Out of abundance of caution, it might be a good idea to add a test for the TOML method of setting also.